pkg/motor/object_builder.go

```diff
@@ -19,3 +21,48 @@ func (mtr *motorNodeImpl) NewObjectBuilder(did string) (*object.ObjectBuilder, e
 	objCli := objectcli.New(schemaImpl, shell.NewShell(mtr.Cosmos.GetIPFSApiAddress()))
 	return object.NewBuilder(schemaImpl, objCli), nil
 }
+
+func (mtr *motorNodeImpl) GetDocument(req mt.GetDocumentRequest) (*mt.GetDocumentResponse, error) {
+	obj, err := mtr.QueryObject(req.GetCid())
+	if err != nil {
+		return nil, err
+	}
+
+	doc := st.NewDocumentFromMap(req.GetCid(), obj)
+	return &mt.GetDocumentResponse{
+		Did:      doc.Did,
+		Status:   200,
+		Document: doc,
+		Cid:      req.GetCid(),
+	}, nil
+}
+
+func (mtr *motorNodeImpl) UploadDocument(req mt.UploadDocumentRequest) (*mt.UploadDocumentResponse, error) {
+	builder, err := mtr.NewObjectBuilder(req.GetDefinition().GetDid())
+	if err != nil {
+		return nil, err
+	}
+
+	builder.SetLabel(req.GetLabel())
+	builder.Set("@did", req.GetDefinition().GetDid())
```

Comment:
`@did` is likely to not be a field in whichever schema is being used. This should cause an error during upload or building.

Reply:
interesting, we could add a clause that '@' prefixed keys are not checked in the schema verification

Reply:
yes

Reply:
we need a way to relate back to the schema